src/Elements/Geometry/Ray.cs

```diff
@@ -80,6 +80,35 @@ public bool Intersects(Triangle tri, out Vector3 result)
             }
         }
 
+        /// <summary>
+        /// Does this ray intersect with the provided GeometricElement? Only GeometricElements with Solid Representations are currently supported, and voids will be ignored.
+        /// </summary>
+        /// <param name="element">The element to intersect with.</param>
+        /// <param name="result">The list of intersection results.</param>
+        /// <returns></returns>
+        public bool Intersects(GeometricElement element, out List<Vector3> result)
+        {
+            List<Vector3> resultsOut = new List<Vector3>();
+            var transformFromElement = new Transform(element.Transform);
+            transformFromElement.Invert();
+            var transformToElement = new Transform(element.Transform);
+            var transformMinusTranslation = new Transform(transformFromElement);
+            transformMinusTranslation.Move(-1 * transformMinusTranslation.Origin); // 
+            var transformedRay = new Ray(transformFromElement.OfPoint(Origin), transformMinusTranslation.OfVector(Direction));
+            //TODO: handle voids
```

Comment:
Is this still `TODO`?

Reply:
Yes — it is contingent on support for void Solids in elements in the first place!